_release/bat/quotas_bat/quotas_bat_test.go

```diff
@@ -96,3 +97,235 @@ func TestQuotas(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func getUsage(qds []bat.QuotaDetails, name string) int {
+	qd := findQuota(qds, name)
+	if qd == nil {
+		return -1
+	}
+
+	value, err := strconv.Atoi(qd.Usage)
+	if err != nil {
+		return -1
+	}
+
+	return value
+}
+
+func checkUsage(qds []bat.QuotaDetails, wl bat.Workload, count int) bool {
+	expectedMem := wl.Mem * count
+	expectedCPU := wl.CPUs * count
+	expectedInstances := count
+
+	actualMem := getUsage(qds, "tenant-mem-quota")
+	actualCPU := getUsage(qds, "tenant-vcpu-quota")
+	actualInstances := getUsage(qds, "tenant-instances-quota")
+	return actualMem == expectedMem &&
+		actualCPU == expectedCPU &&
+		actualInstances == expectedInstances
+}
+
+func getContainerWorkload(ctx context.Context, tenantID string) (bat.Workload, error) {
+	const testContainerWorkload = "Debian latest test container"
+	return bat.GetWorkloadByName(ctx, tenantID, testContainerWorkload)
+}
+
+// Test reporting of instance usage
+//
+// Starts 3 copies of a workload and checks that the usage matches
+// 3 * workload defaults
+//
+// Gets a workload, launches 3 instances from that workloads, gets the
+// quota and usage information and then checks that the usage is as
+// expected. Deletes the instances and checks the usage reflects that.
+func TestInstanceUsage(t *testing.T) {
+	ctx, cancelFunc := context.WithTimeout(context.Background(), standardTimeout)
+	defer cancelFunc()
+	tenants, err := bat.GetUserTenants(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(tenants) < 1 {
+		t.Fatal("Expected user to have access to at least one tenant")
+	}
+	tenantID := tenants[0].ID
+
+	wl, err := getContainerWorkload(ctx, tenantID)
+	if err != nil {
+		t.Skip()
+	}
+
+	instances, err := bat.LaunchInstances(ctx, "", wl.ID, 3)
+	if err != nil {
+		t.Error(err)
+	}
+
+	scheduled, err := bat.WaitForInstancesLaunch(ctx, "", instances, false)
+	if err != nil {
+		t.Fatalf("Instance %s did not launch: %v", instances[0], err)
```

Comment:
Also, as this is a Fatal, it means the instances that did launch will not get cleaned up.